lib/s3middleware/lifecycleHelpers/LifecycleUtils.ts

```diff
@@ -207,13 +207,15 @@ export default class LifecycleUtils {
                 // Names are long, so obscuring a bit
                 const ncve = 'NoncurrentVersionExpiration';
                 const ncd = 'NoncurrentDays';
+                const nncv = 'NewerNoncurrentVersions';
 
                 if (!store[ncve]) {
                     store[ncve] = {};
                 }
                 if (!store[ncve][ncd] || rule[ncve][ncd] < store[ncve][ncd]) {
                     store[ncve].ID = rule.ID;
                     store[ncve][ncd] = rule[ncve][ncd];
+                    store[ncve][nncv] = rule[ncve][nncv];
```

Comment:
Do we support `NewerNoncurrentVersions` without `NoncurrentDays`, like AWS seems to support? If so we should separate the setting of `nncv` specifically (otherwise, it looks fine as is to me, we will set the value of `NewerNoncurrentVersions` from the rule with the most constrained `NoncurrentDays` rule).

Reply:
@jonathan-gramain 

testing AWS S3 behavior, `NoncurrentDays` is a required value; 

there looks to be some inconsistency in AWS documentation:
* https://docs.aws.amazon.com/AmazonS3/latest/API/API_NoncurrentVersionExpiration.html
* other service integration https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig-rule-noncurrentversionexpiration.html

in the S3 console, this is a required value